roles/awx/tasks/install.yml

```diff
@@ -0,0 +1,83 @@
+---
+# Based on https://github.com/ansible/awx/blob/1.0.3/installer/openshift/tasks/main.yml
+
+- name: Ensure AWX Project Exists
+  oc_project:
+    name: "{{ awx_project }}"
+
+# This secret is created by postgresql-persistent template
+- name: Read Existing Postgres Secret
+  oc_secret:
+    state: list
+    namespace: "{{ awx_project }}"
+    name: postgresql
+    decode: true
+  register: awx_postgres_secret
+  no_log: true
+
+- name: Get or Generate AWX Postgres Password
+  set_fact:
+    pg_password: "{{ awx_postgres_secret.results.decoded['database-password'] | default(16 | lib_utils_oo_random_word) }}"
+  no_log: true
+
+- name: Deploy and Activate Postgres
+  shell: >-
+    {{ openshift_client_binary }} project {{ awx_project }};
+    {{ openshift_client_binary }} process -n openshift postgresql-persistent
```

Comment:
Could `oc_process` module be used here?

Reply:
I tried initially and to be honest I don't fully remember why I dropped it :-)
Hmm, from oc_process.py:
> in conjunction with 'create: True' the process will be piped to | oc create -f

That doesn't sound idempotent.  I guess I need to wrap it in a condition to create unless exists?
Then what if you upgrade openshift and the builtin `postgresql-persistent` template changes but you won't pick up changes.  So I think `| oc apply` is better?
Or maybe I'm looking at this wrong, and upgrading a postgres in place is actually undesired?

Reply:
Do you mean the oc_apply role instead of oc apply command?

Reply:
Yes, `oc_apply` role would make it cleaner.

Reply:
Switched from j2 to an openshift template.

- Used one `oc_configmap` because the main payload is a fixed python file (getting param by env var instead of textual templating) and I felt it's much cleaner to just have it as a file.

The two existing `oc_apply` wrappers I found (`roles/openshift_metrics/tasks/oc_apply.yaml`, `roles/openshift_provisioners/tasks/oc_apply.yaml`) added value is reporting "changed" correctly. They seem to expect a single object, as they check resourceVersion before/after.  I'd have to complicate that approach to iterate the multiple objects the template creates.
=> Instead, I'm running `oc apply` directly, formulated a `changed_when` that looks at output.  Tested with/without change and with several errors (both in template and apply stage).